spring-cloud-function-integration/src/main/java/org/springframework/cloud/function/integration/dsl/FunctionFlowAutoConfiguration.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2023-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.function.integration.dsl;
+
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.cloud.function.context.FunctionCatalog;
+import org.springframework.cloud.function.context.config.ContextFunctionCatalogAutoConfiguration;
+import org.springframework.context.annotation.Bean;
+
+/**
+ * The auto-configuration to expose a {@link FunctionFlowBuilder} bean
+ * based on the auto-configured {@link FunctionCatalog}.
+ *
+ * @author Artem Bilan
+ *
+ * @since 4.0.3
+ */
+@AutoConfiguration(after = ContextFunctionCatalogAutoConfiguration.class)
+public class FunctionFlowAutoConfiguration {
+
+	@Bean
+	FunctionFlowBuilder functionFlowBuilder(FunctionCatalog functionCatalog) {
+		return new FunctionFlowBuilder(functionCatalog);
+	}
```

Comment:
It might be a good idea to introduce a simple autoconfig test (using `ApplicationContextRunner`) .

Reply:
And what that test suppose to verify?
We really don't develop here Spring Boot stuff.
We do our functions logic and verify it using `@SpringBootTest` which really cover this auto-config as well.

Reply:
This comment was contingent on adding other auto-configured beans / conditions, etc.. which are not needing to do now so the autoconfig test would not be valuable now.